Relax the requirment for the latest flex version.
Remove the magic compiler flags.
Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -98,13 +98,13 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 check_pie_supported() # opportunistically enable PIE
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-	set(CMAKE_CXX_FLAGS_DEBUG "-Og -ggdb")
-	set(CMAKE_CXX_FLAGS_RELEASE "-O3")
-	set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -ggdb")
-	set(CMAKE_CXX_FLAGS_MINSIZEREL "-Os")
-	set(CMAKE_CXX_FLAGS_SANITIZE "-O1 -fno-omit-frame-pointer -fno-optimize-sibling-calls")
+	set(CMAKE_CXX_FLAGS_DEBUG "")
+	set(CMAKE_CXX_FLAGS_RELEASE "")
+	set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "")
+	set(CMAKE_CXX_FLAGS_MINSIZEREL "")
+	set(CMAKE_CXX_FLAGS_SANITIZE "")
 	if ("${SANITIZE}" MATCHES "memory")
-		set(CMAKE_CXX_FLAGS_SANITIZE "${CMAKE_CXX_FLAGS_SANITIZE} -fsanitize-memory-track-origins")
+		set(CMAKE_CXX_FLAGS_SANITIZE "")
 	endif()
 	set(no_abc_options
 		"$<$<AND:$<NOT:$<BOOL:$<TARGET_PROPERTY:YOSYS_IS_ABC>>>,$<CONFIG:Sanitize>>:-fsanitize=${SANITIZE}>"
@@ -181,7 +181,7 @@ if (APPLE AND (NOT DEFINED YOSYS_ENABLE_HOMEBREW OR YO
 	use_homebrew()
 endif()
 
-find_package(FLEX 2.6)
+find_package(FLEX 2.5.39)
 set_package_properties(FLEX PROPERTIES
 	URL "https://github.com/westes/flex"
 	DESCRIPTION "The Fast Lexical Analyzer"
